### PR TITLE
Change all of body system to use entities and components

### LINKED ISSUE #1753: Female body part sprites are currently broken
Because of how body parts are mapped to one rsi state in the body parts YAMLs, the body parts cannot reliably change appearance between their male and female versions. This was previously done with the humanoid appearance component changing the sprite dynamically, but can be better accomplished now by creating separate body parts and presets for male and female humans.

This should ideally be fixed by creating separate body parts for the female variants, and a separate human preset to include them.

### LINKED ISSUE #1757: Body networks currently serve no purpose
All they do is act as a proxy for adding/removing the relevant component, for example the lung component when the lungs installed or taken out. They should either take the role of the relevant components, or act as a linking mechanism for different types of organs to coordinate them.

Related to #1762

### LINKED ISSUE #1758: Body system should be tested under the entity/component model
Currently, body parts and organs are only entities with components when dropped. This makes it very difficult and annoying to work with them in the current ECS, as any interactions with other components need to first be routed through BodyManagerComponent. This also means that any improvements to the ECS don't directly apply to body system as it is isolated, and any improvements to the body system do not apply back to the general ECS. Reinventing the wheel is a problem.

The concerns with transitioning it to use the EC model are performance, and primarily memory, so they should be bench-marked accordingly in the associated PR.

image

### LINKED ISSUE #1760: Add damaging specific body parts
Right now you cannot damage specific parts of a body, such as the hand when touching a lightbulb. The framework for it is already there as each body part has its own DamageContainer. BodyManagerComponent's damage methods would need to be cahnged to check for BodyManagerHealthChangeParams when called, and if so distribute the damage across its limbs. This would also require the removal of DamageContainer from the body as the damage would be isolated to the parts instead.
More elaborate methods could also be added, just as parameters that spread the damage across different body parts with a bias.

### LINKED ISSUE #1761: Damageable component is currently not able to be predicted by the client
Its NetID was even removed during the great body system transition. This means that under high ping any code that causes damage to anynthing will not be capable of being smooth for the client.

### LINKED ISSUE #1762: Removing a body network currently removes the relevant entity's components
Each body has a dictionary of networks which it manages, however these networks remove their associated components when they themselves are removed. For example, when the circulatory network is removed, it removes BloodstreamComponent. In real life however, your bloodstream's contents don't instantly vanish when your heart is removed and your entire respiratory system doesn't disappear when your lungs are removed.

Related to #1757

### LINKED ISSUE #1763: Body manager component needs to be refactored and split off
1016 lines of code for a body class is not ok.
It can be split off into separate or partial classes, for example one that manages body parts exclusively.
Its methods are also named and designed inconsistently, with different arguments for methods that accomplish the same thing, for example removing body parts.
It should also be changed to use interfaces instead.

### LINKED ISSUE #2171: HumanoidCharacterAppearance SkinColor and EyeColor are not used anywhere
They can't be set in the character editor and aren't read anywhere in the code to actually set the skin color or eye color of a character.

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Fixes #1753 
Fixes #1757 
Closes #1758 
Fixes #1760 
Fixes #1761 
Fixes #1762 
Fixes #1763 
Fixes #2171 

~~Requires https://github.com/space-wizards/RobustToolbox/pull/1291~~
~~Requires https://github.com/space-wizards/RobustToolbox/pull/1299~~
~~Requires https://github.com/space-wizards/RobustToolbox/pull/1303~~
~~Requires https://github.com/space-wizards/space-station-14/pull/2172~~

Body networks are gone.
Body properties are now components.
Half the codebase is gone.

TODO: Cleanup, surgery and using HumanoidAppearance.EyeColor.

### CODE REVIEW

**Comment by GlassEclipse** [comment]
👀

**Comment by RemieRichards** [comment]
> does anyone even read these

👀

**Comment by RemieRichards** [comment]
> life is fleeting

Just doing it again to annoy you smug 👀

**Comment by GlassEclipse** [comment]
Out of curiosity how will you be doing things like a body-wide bloodstream or body-wide electric system without bodynetworks

**Comment by GlassEclipse** [comment]
Also does this mean that humans will have like 7 components, each representing a different limb
Cause that was what I was trying to avoid

**Comment by ancientpower** [comment]
It means humans are going to have ~20 entities attached to them representing limbs/organs. Unless someone produces benchmarks showing that it causes a massive performance hit I don't see why that should be avoided considering the massive convenience factor it introduces. If it does cause a massive performance hit then that would be pretty concerning considering one player is probably going to produce way more than 20 entities a round on average.

**Comment by GlassEclipse** [comment]
Yeah that is pretty much what I wanted to avoid... I have to imagine entity overhead contains a lot of shit, even more than components, but I'm not doing it anymore, so it's up to mr. smug I suppose. This way does simplify things a bit.

**Comment by F77F** [comment]
lol i know that feel

**Comment by DrSmugleaf** [comment]
Ready for review.

**Review by Zumorica — APPROVED** [review]
LGTM. Feel free to merge this after you've addressed my one and only review.

File: Content.Client/GameObjects/EntitySystems/VerbSystem.cs [hunk 1]
```diff
@@ -129,6 +130,11 @@ private bool OnOpenContextMenu(in PointerInputCmdHandler.PointerInputCmdArgs arg
                     continue;
                 }
 
+                if (entity.GetAllComponents<IShowContextMenu>().Any(s => !s.ShowContextMenu(playerEntity)))
+                {
+                    continue;
+                }
+
```
[the remark this answers is not in the document]
Comment (DrSmugleaf) [reply Content.Client/GameObjects/EntitySystems/VerbSystem.cs]: Done.

**Pull request merged**